### PR TITLE
Add support for including code files with functions

### PR DESCRIPTION
**Pull request by davidebbo**
Mostly for @ahmelsayed, as this is going into his branch and not master.

The idea is to support:

```
{
  "config": {

  },
  "files":{
      "foo.js": "bar",
      "hello.js": "This is a test"
  }
}
```

### CODE REVIEW

File: Kudu.Contracts/Jobs/FunctionEnvelope.cs [hunk 1]
```diff
@@ -33,5 +33,8 @@ public class FunctionEnvelope : INamedObject
 
         [JsonProperty(PropertyName = "config")]
         public JObject Config { get; set; }
+
+        [JsonProperty(PropertyName = "files")]
+        public JObject Files { get; set; }
```
Comment (ahmelsayed) [inline Kudu.Contracts/Jobs/FunctionEnvelope.cs]: Why not `Dictionary<string, string>`?

Comment (ahmelsayed) [inline Kudu.Contracts/Jobs/FunctionEnvelope.cs]: No we don't have any unite tests yet :) I know I'll add some

**Comment by ahmelsayed** [comment]
No we don't have any unite tests yet :) I know I'll add some

File: Kudu.Contracts/Jobs/FunctionEnvelope.cs (hunk 1 above)
Comment (davidebbo) [inline Kudu.Contracts/Jobs/FunctionEnvelope.cs]: Changed as suggested.

**Comment by davidebbo** [comment]
I also added ARM support to the controller.

**Comment by suwatch** [comment]
just happen to glance around the implementation.   Unrelated to this change, when doing GetTemplatesFromGithub or DownloadFromGithub, we should have tracing around it.  It will help us diagnose issues.

**Comment by davidebbo** [comment]
We'll probably be yanking that GitHub logic and manage the templating from the outside instead (making it more of a normal creation).

**Pull request merged**